public/app/features/alerting/unified/components/receivers/form/CloudReceiverForm.tsx

```diff
@@ -44,9 +47,9 @@ export const CloudReceiverForm = ({ existing, alertManagerSourceName, config }:
     return cloudReceiverToFormValues(existing, cloudNotifierTypes);
   }, [existing]);
 
-  const onSubmit = (values: ReceiverFormValues<CloudChannelValues>) => {
+  const onSubmit = async (values: ReceiverFormValues<CloudChannelValues>) => {
     const newReceiver = formValuesToCloudReceiver(values, defaultChannelValues);
-    dispatch(
+    await dispatch(
```

Comment:
Why do we need to add these awaits before dispatching `updateAlertManagerConfigAction`?

Reply:
Because `onSubmit` is an async action (we send a request to the server) and for `react-hook-form` to handle `isSubmitting` state correctly, all async actions inside the `onSubmit` function need to be properly awaited.
Thanks to this change we can use `isSubmitting` property of `formState` instead of doing error-prone manual request state checking (https://github.com/grafana/grafana/pull/72021/files#diff-fdd6d5142b270599bec93afc64eb1fa15ad762a6cb8163ff8b46c4ee1ff3310fR80)

Reply:
Nice! 🙌